### PR TITLE
Clarify and prettify initial SQL

### LINKED ISSUE #150: More informative default query
Currently the default query produced by Dashboard is pretty simple:

``` sql
SELECT "relate", count(*) FROM "census_small" GROUP BY 1
```

I wonder if we could make a somewhat more informative query. The idea would be to pick one column of each type (numeric, text, datetime), and make a query with all three (or in this case two, since there isn't a date column in this table, plus add some basic hints:

``` sql
SELECT "relate",
       "age",
       count(*) 
FROM "census_small" 
GROUP BY 1,2 


-- HINTS 
-- Change, add, or remove columns as desired
-- Text columns like "relate" can be masked:
  -- substring("relate",1,2)
-- Numeric columns like "age" can be binned:
  -- diffix.floor_by("age",10)
-- Sum numeric columns with sum(column)
-- Learn more at some_url
```

### PR DESCRIPTION
**Pull request by pdobacz**
Closes #150 insofar as the changes not overlapping with #77 go.

The third commit is a bit of a refactor, which produces a new module, which is importable both from renderer and main process modules - just generic, shared, utility code. Not sure if this is nice enough by standards for an electron app.

### CODE REVIEW

File: src/main/metabase/api.ts [hunk 1]
```diff
@@ -248,11 +274,15 @@ export async function buildInitialQueries(
   // Picks some fields which aren't AIDs to put in the GROUP BY.
   const nonAidField = fields.find((field) => !aidColumns.includes(field.name));
 
-  const columnSQL = nonAidField ? `"${nonAidField.name}",` : '';
-  const groupBySQL = nonAidField ? 'GROUP BY 1' : '';
-  const query = `SELECT ${columnSQL} count(*) FROM "${tableName}" ${groupBySQL}`;
+  const columnSQL = nonAidField ? `${postgresQuote(nonAidField.name)},` : '';
+  const groupBySQL = nonAidField ? `GROUP BY ${postgresQuote(nonAidField.name)}` : '';
+  const query = safeSQLFormat(`SELECT ${columnSQL} count(*) FROM ${postgresQuote(tableName)} ${groupBySQL}`);
```
Comment (edongashi) [inline src/main/metabase/api.ts:279]: We are not accepting dynamic user input that we need to format. We can yielded whatever SQL format we want.

What's the advantage compared to something like this?:
```js
[
  `SELECT ${columns}`,
  'FROM ${table}',
  'GROUP BY 1, 2'
].join('\n')
```

Comment (pdobacz) [reply src/main/metabase/api.ts:279]: It looks a bit more like the indented version Paul suggested, which I don't want to implement by hand, but yours also works. I guess we can get rid of the dependency for now and stick to the simple format.

File: src/main/metabase/api.ts [hunk 2]
```diff
@@ -248,11 +264,15 @@ export async function buildInitialQueries(
...
-  const columnSQL = nonAidField ? `"${nonAidField.name}",` : '';
-  const groupBySQL = nonAidField ? 'GROUP BY 1' : '';
-  const query = `SELECT ${columnSQL} count(*) FROM "${tableName}" ${groupBySQL}`;
+  const columnSQL = nonAidField ? `${postgresQuote(nonAidField.name)},` : '';
+  const groupBySQL = nonAidField ? `GROUP BY ${postgresQuote(nonAidField.name)}` : '';
+  const query = [`SELECT ${columnSQL} count(*)`, `FROM ${postgresQuote(tableName)}`, `${groupBySQL}`].join('\n');
+  const commentedQuery = `
+${query}
+
+${sqlHint}`;
```
Comment (edongashi) [inline src/main/metabase/api.ts:273]: The following should better control where we want (and how many)  newlines:

```suggestion
  const commentedQuery = `${query}\n\n${sqlHint}`;
```

File: src/shared/utils.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { postgresReservedKeywords } from '../constants';
```
Comment (edongashi) [inline src/shared/utils.ts:1]: We can now move `constants` to `shared/constants` if you so desire.

**Review by edongashi — APPROVED** [review]

File: src/shared/utils.ts (hunk 3 above)
Comment (pdobacz) [reply src/shared/utils.ts:1]: but I guess without exporting at `shared/index.ts`?

Comment (edongashi) [reply src/shared/utils.ts:1]: Hmm, yes. I have some regrets about exporting everything from index files.

**Review by edongashi — APPROVED** [review]

**Pull request merged**